tcclient/http.go

```diff
@@ -69,12 +82,9 @@ func (connectionData *ConnectionData) APICall(payload interface{}, method, route
 		if reflect.ValueOf(payload).IsValid() && !reflect.ValueOf(payload).IsNil() {
 			ioReader = bytes.NewReader(jsonPayload)
 		}
-		u, err := url.Parse(connectionData.BaseURL + route)
+		u, err := setURL(connectionData, route, query)
 		if err != nil {
-			return nil, nil, fmt.Errorf("apiCall url cannot be parsed: '%v', is your BaseURL (%v) set correctly?\n%v\n", connectionData.BaseURL+route, connectionData.BaseURL, err)
-		}
-		if query != nil {
-			u.RawQuery = query.Encode()
+			return nil, nil, fmt.Errorf("apiCall url cannot be parsed:\n%v\n", err)
```

Comment:
Do we need to prefix the error with anything or couldn't we just return the error that setUrl returned?